### PR TITLE
Mentioned that partial methods cannot be constructors

### PR DESCRIPTION
**Pull request by SteveDunn**
## Summary

Mentioned that partial methods cannot be constructors

### CODE REVIEW

**Comment by SteveDunn** [comment]
@BillWagner - this PR is a result from experimenting with 'value objects/invariants' - I'd love your comments and thoughts on this thread in a language proposal: https://github.com/dotnet/csharplang/discussions/5075#discussioncomment-1856470

**Comment by BillWagner** [comment]
Hi :wave: @SteveDunn 

I like this change. I would like to wait to merge this PR to address the [other restrictions](https://docs.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-9.0/extending-partial-methods#partial-on-all-members) that still exist for partial methods.  The language in the feature spec is a bit vague, so we'll have to look through the LDM notes on partial methods to be sure what was defined. I *think* that none of these are allowed to be partial:

- constructors
- finalizers
- operators
- properties (I think the parser interprets them as auto-implemented properties).

File: docs/csharp/language-reference/keywords/partial-method.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ A partial method has its signature defined in one part of a partial type, and it
 
 - Signatures in both parts of the partial type must match.
 
+- Signatures are not constructors.
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/partial-method.md:19]: Given all the restrictions, I'd write this as follows:

```suggestion
The `partial` keyword isn't allowed on constructors, finalizers, overloaded operators, property declarations, or event declarations. 
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for making this addition @SteveDunn 

After checking the LDM notes, I made a suggestion to update the restrictions. Once you accept that, it's ready to merge.

Thanks again.

**Comment by SteveDunn** [comment]
Many thanks Bill. Accepted that change.

**Review by BillWagner — APPROVED** [review]

File: docs/csharp/language-reference/keywords/partial-method.md [hunk 2]
```diff
@@ -16,6 +16,8 @@ A partial method has its signature defined in one part of a partial type, and it
 
 - Signatures in both parts of the partial type must match.
 
+The `partial` keyword isn't allowed on constructors, finalizers, overloaded operators, property declarations, or event declarations.
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/partial-method.md:19]:
```suggestion
The `partial` keyword isn't allowed on constructors, finalizers, overloaded operators, property declarations, or event declarations.
```

**Pull request merged**